js/chatbot/shared/ChatBot.svelte

```diff
@@ -197,16 +201,13 @@
 
 	function is_last_bot_message(
 		messages: NormalisedMessage[],
-		total_length: number
+		all_messages: NormalisedMessage[]
 	): boolean {
 		const is_bot = messages[messages.length - 1].role === "assistant";
 		const last_index = messages[messages.length - 1].index;
-		let is_last;
-		if (Array.isArray(last_index)) {
-			is_last = 2 * last_index[0] + last_index[1] === total_length - 1;
-		} else {
-			is_last = last_index === total_length - 1;
-		}
+		const is_last =
+			JSON.stringify(last_index) ===
+			JSON.stringify(all_messages[all_messages.length - 1].index);
```

Comment:
Sorry can you explain what's happening here @freddyaboulton? Why are we stringifying what should just be numerical indices?

Reply:
In the tuples case the index is an array of the form (message_index, 0 if user or 1 if bot). And in javascript comparing two arrays will be false even if they have the same elements. So the code in the PR will work for either case.

Can't wait for 6.0 😅